internal/html/scrape/schema/json-ld/recipe.go

```diff
@@ -110,11 +136,38 @@ func addSchemaCtx(v any) {
 func findRecipeNode(nodes []any) (map[string]any, bool) {
 	for _, node := range nodes {
 		if m, ok := node.(map[string]any); ok {
-			if m[typeKey] == recipeType {
+			str, arr, err := ConvertToStringOrArray(m[typeKey])
+			if err != nil {
+				return nil, false
+			}
+
+			if arr != nil && ArrayContains(arr, recipeType) {
+				return m, true
+			} else if str != "" && str == recipeType {
 				return m, true
 			}
 		}
 	}
 
 	return nil, false
 }
+
+func ConvertToStringOrArray(data any) (string, []interface{}, error) {
```

Comment:
We should not be exporting functions for the sole purpose of using them as helpers in a test.

Reply:
will fix